src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const act of action.keysToRemove) {
+          delete newState[act];
+        };
+        break;
+
+      case 'clear':
```

Comment:
Don't forget about default case
It should be in all switch / case statement
You can just add break operator here